parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -1658,11 +1659,15 @@ public void end(Map<String, String> extraMetaData) throws IOException {
 
   @Override
   public void close() throws IOException {
+    if (closed) {
+      return;
+    }
     try (PositionOutputStream temp = out) {
       temp.flush();
       if (crcAllocator != null) {
         crcAllocator.close();
       }
+      closed = true;
```

Comment:
Should we put this in the finally block just in case of any exception?

Reply:
fixed

Reply:
Putting it in finally block makes it be not retryable.

Reply:
> Putting it in finally block makes it be not retryable.

That was also my initial thought. I was actually torn between these two choices, but I suppose it's rare for people to actually retry failed close operation. Feedback is welcome.

Reply:
> it's rare for people to actually retry failed close operation

I agree with you, it's rare.
But if someone retries, it will not work as expected -- file is not closed as expected. This is not user friendly, right?

Reply:
I notice that `try {} finally { closed = true; }` is common in parquet writer code base, so it' ok to follow up.